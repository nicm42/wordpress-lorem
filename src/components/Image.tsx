import PropTypes from 'prop-types';

interface IPhotoProps {
  photo: any;
}

const Image = ({ photo }: IPhotoProps) => {
  return (
    <div className="text-center">
      <img className="img-fluid" src={photo.link} alt={photo.alt} />
      <p className="small">
        Photo by <a href={photo.userLink}>{photo.userName}</a> on{' '}
        <a href="https://unsplash.com/?utm_source=wordpress_lorem&utm_medium=referral">
          Unsplash
        </a>
      </p>
    </div>
  );
};

Image.propTypes = {
  photo: PropTypes.shape({
    link: PropTypes.string,
    alt: PropTypes.string,
    userName: PropTypes.string,
    userLink: PropTypes.string,
  }),
};

export default Image;
