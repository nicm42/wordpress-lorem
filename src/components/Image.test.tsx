import { render, screen } from '@testing-library/react';
import Image from './Image';
import testImage from '../test-image2.json';

describe('Image test', () => {
  it('renders without crashing', () => {
    render(<Image photo={testImage} index={1} />);
    const image = screen.getByRole('img');
    expect(image).toBeInTheDocument();
  });

  it('has the correct src and alt', () => {
    render(<Image photo={testImage} index={1} />);
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute(
      'src',
      'https://images.unsplash.com/photo-1571622840901-92ae138bd36e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyNDc4NTd8MHwxfHJhbmRvbXx8fHx8fHx8fDE2MjY3NjA2NTU&ixlib=rb-1.2.1&q=80&w=200'
    );
    expect(image).toHaveAttribute(
      'alt',
      'clear glass cake with white and red strawberries'
    );
  });

  it('has the photographer information present', () => {
    render(<Image photo={testImage} index={1} />);
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(2);
    expect(links[0]).toHaveAttribute(
      'href',
      'https://unsplash.com/@jjy95?utm_source=wordpress_lorem&utm_medium=referral'
    );
    expect(links[1]).toHaveAttribute(
      'href',
      'https://unsplash.com?utm_source=wordpress_lorem&utm_medium=referral'
    );
    const linkText1 = screen.getByText('jjy95');
    expect(linkText1).toBeInTheDocument();
    const linkText2 = screen.getByText('Unsplash');
    expect(linkText2).toBeInTheDocument();
  });
});
