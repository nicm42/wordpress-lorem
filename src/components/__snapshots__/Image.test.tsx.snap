// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests renders the images correctly 1`] = `
<figure
  className="figure text-center float-md-end px-3"
>
  <img
    alt="clear glass cake with white and red strawberries"
    className="figure-img img-fluid"
    loading="lazy"
    src="https://images.unsplash.com/photo-1571622840901-92ae138bd36e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyNDc4NTd8MHwxfHJhbmRvbXx8fHx8fHx8fDE2MjY3NjA2NTU&ixlib=rb-1.2.1&q=80&w=200"
  />
  <figcaption
    className="figure-caption"
  >
    Photo by 
    <a
      href="https://unsplash.com/@jjy95?utm_source=wordpress_lorem&utm_medium=referral"
    >
      jjy95
    </a>
     on
     
    <a
      className="link-primary"
      href="https://unsplash.com?utm_source=wordpress_lorem&utm_medium=referral"
    >
      Unsplash
    </a>
  </figcaption>
</figure>
`;
